x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -169,11 +170,37 @@ public void onFailure(Exception e) {
         );
 
         // Tell the job tracker to refresh the memory requirement for this job and all other jobs that have persistent tasks
+        ActionListener<EstimateMemoryUsageAction.Response> estimateMemoryUsageListener = ActionListener.wrap(
+            estimateMemoryUsageResponse -> {
+                // Validate that model memory limit is sufficient to run the analysis
+                if (configHolder.get().getModelMemoryLimit()
+                    .compareTo(estimateMemoryUsageResponse.getExpectedMemoryUsageWithOnePartition()) < 0) {
+                    ElasticsearchStatusException e =
+                        new ElasticsearchStatusException(
+                            "Cannot start data frame analytics [" + request.getId() + "] because the configured memory limit [" +
```

Comment:
Usually we do not report the name of the resource here because it is a direct response to the API call and the user will be in the context to know which API failed for which resource. We could shorten to:

```
Cannot start because the configured memory...
```

Reply:
Done.